src/ProgressOnderwijsUtils/Data/CascadedDelete.cs

```diff
@@ -170,19 +158,38 @@ void DeleteKids(DatabaseDescription.Table table, ParameterizedSql tempTableName,
                     () => {
                         var nrRowsToDelete = SQL($"select count(*) from {tempTableName}").ReadScalar<int>(conn);
                         log($"Delete {nrRowsToDelete} from {table.QualifiedName}...");
+
+                        ParameterizedSql DeletionQuery(ParameterizedSql outputClause)
+                            => SQL($@"
+                                delete pk
+                                {outputClause}
+                                from {table.QualifiedNameSql} pk
+                                join {tempTableName} tt on {ttJoin};

```

Comment:
ik weet niet of het *beter* is, maar als je zo'n setup had...

```SQL
{outputPrepStatement}

delete pk
{outputColumns} {outputInto}
from {table.QualifiedNameSql} pk
join {tempTableName} tt on {ttJoin};

{outputResultsStatement}
```

dan zou je de losse stukjes in losse ifjes kunnen doen, en was het mogelijk natuurlijker om bijv. outputColumns hetzelfde te houden in de tak zoder en met trigger, en was het ook mogelijk om DeletionQuery in DeletionExecution te inlinen zonder duplicatie.

Maar mischien is dat juist te clever.

Reply:
Ik vind persoonlijk mijn versie nu leesbaarder. beetje look om oud ijzer toch?